src/lib/select/select.ts

```diff
@@ -512,6 +513,14 @@ export class MatSelect extends _MatSelectMixinBase implements AfterContentInit,
       this._resetOptions();
       this._initializeSelection();
     });
+
+    this._triggerRect = this.trigger.nativeElement.getBoundingClientRect();
```

Comment:
It seems like you're recalculating the `_triggerRect` here and right afterwards inside `_initializePanel`. We should be able to get away with only doing it in `_initializePanel`.

Reply:
The trigger's bounding rectangle may change after the panel is initialized. 
The panel is only initialized once when the component is loaded, but if the trigger's rect changed after that, we still need to get the correct `scrollTop`, `offsetX`, `offsetY` based on the new values.